pkg/controller/render/machine_config_to_ignition.go

```diff
@@ -0,0 +1,107 @@
+package render
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	baseutil "github.com/coreos/butane/base/util"
+	"github.com/coreos/ignition/v2/config/util"
+	ign3types "github.com/coreos/ignition/v2/config/v3_2/types"
+	mcSpecv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/test/helpers"
+
+	"gopkg.in/yaml.v3"
+)
+
+const MCDContentPath = "/etc/machine-config-daemon/mcd_content.json"
+
+type MCDContent struct {
+	KernelArguments []string `json:"kernelArguments"`
+	FIPS            bool     `json:"fips"`
+}
+
+func MachineConfigToIgnition(mcSpec *mcSpecv1.MachineConfigSpec) (ign3types.Config, error) {
+	// Config
+	ignition, err := ctrlcommon.ParseAndConvertConfig(mcSpec.Config.Raw)
+	if err != nil {
+		return ign3types.Config{}, fmt.Errorf("parsing Ignition config failed: %w", err)
+	}
+
+	var treeFile TreeFile
+	// Extensions
+	supportedExtensions := ctrlcommon.GetSupportedExtensions()
+	for _, mcSpecExtension := range mcSpec.Extensions {
+		treeFile.Packages = append(treeFile.Packages, supportedExtensions[mcSpecExtension]...)
+	}
+
+	// KernelType
+	if ctrlcommon.CanonicalizeKernelType(mcSpec.KernelType) == ctrlcommon.KernelTypeRealtime {
```

Comment:
Feels like this one should be a `switch` or so and we explicitly no-op on the default kernel type?  But not a blocker.

Reply:
Fixed, should I return err for default?